MyChats Revamp.md (#370)
hellohaptik/docs_haptik: docs/agent-chat/for-chat-agents.md

## Logging in

The chat agent gets an email with credentials to the **Smart Agent Chat** tool. 

Using the credentials, a chat agent logs in and clicks on **Smart Agent Chat**. Now, the chat agent can essentially click on the **My Chats** tab i.e. the first option depicted in the left-hand menu.

## How to use My Chats?

SCREENSHOT

In this tab the chat agent can start taking chats assigned to him/her personally. The screen is divided into multiple sections. We will call it LHS, Centre and RHS for ease of understanding.

### Starting with the Left Hand Side of the screen - 

LHS SCREENSHOT

A chat agent can:
  * View agent's display name

>   * View Agent Status - Online or Offline

Reviewer: Offline -- might need to re-named

Reviewer: might need more information here, we should create a new page for this.

Reviewer: @medhaanand

Author: Can you tell me what all an agent can do in online state and what all agent cant do in offline state @deepwired ?

  * View his live chats stats (Completed, Waiting for user, My Queue, Team Queue)
  * View his active chats

Name and Status SCREENSHOT

> #### Agent Bar

Reviewer: We might need to rename this

> #### Agent Bar

Reviewer: Agent Details or Agent Status, or Agent name

Reviewer: @medhaanand

> #### Agent Bar

Author: ```suggestion
#### Agent Details
```

The agent will be able to view his name and also his current status. 

POP SCREENSHOT

Agent can also choose the change the status by clicking on **"Go Online"** or **"Go Offline"**. Before chnaging the status, a pop up would come to confirm the change.

#### Live Chat Stats

> This section will give all numbers important for the agent.

Reviewer: Incoming, Existing and completed conversation ka information

Reviewer: we should change the text to portray the above description. @medhaanand

Live chats SCREENSHOT

- **Completed**: Number of chats completed today, as per the agents timezone.

> - **Waiting for user**: Number of chats currently in waiting state of this agent.

Reviewer: waiting chats, instead of waiting for user chats.

Reviewer: renaming @medhaanand

> - **Agent queue**: Chats that have been manually assigned to the agent, or **Waiting for user** chats that has got a user reply

Reviewer: reassing chat --- instaed of assign chat

Reviewer: small text change @medhaanand

- **Team Queue**: Chats that are currently waiting to be taken up by the team. 

> If an agent is part of multiple teams, we show the combined number for team queue.

#### Your Active Chats

This section will show all chats that are currently assigned to the agent.

Your active chats SCREENSHOT (Will all types of tags)

We show the following details -

> - **User name**: This is the name of the user. If the user has provided his name, that would show up or it will be **Guest User**.

Reviewer: team name is also present

Reviewer: description update @medhaanand

> - **Last message time**: This will either be the first user message time (among consecutive user messages), or the latest agent reply time, whichever is earlier.

Reviewer: call this message time, instead of last message time.

Reviewer: separately mention unread and read ka timestamp scenarios

Reviewer: @medhaanand more info present on the design figma link

- **Waiting longest**: This is a tag used for chats which have not had a reply for over 5 mins, or the longest waiting chat.
- **New Chat**: This is a tag whenever a new chat comes in.

> - **Number tab**: This is a tag showing number of messages that the agent has not yet read or responded to.

Reviewer: 2 versions to this...

user unread or read messages... Tanvi was explaining something.

> ## Logging in

Reviewer: lets remove this logging in thing, i don't think its required.

Author: Done

The chat agent gets an email with credentials to the **Smart Agent Chat** tool. 

Using the credentials, a chat agent logs in and clicks on **Smart Agent Chat**. Now, the chat agent can essentially click on the **My Chats** tab i.e. the first option depicted in the left-hand menu.

## How to use My Chats?

![MyChatsScreen1](assets/mychatsscreen.png)

In this tab, an agent can start taking chats assigned to him/her personally. The screen is divided into multiple sections. We will call it LHS, Centre and RHS for ease of understanding.

### Left Hand Side of the screen

![MyChatsScreen2](assets/leftside.png)

A chat agent can:
  * View agent's display name
  * View agent's status - Online or Offline
  * View his live chats stats (Completed, Waiting for user, My Queue, Team Queue)
  * View his active chats

#### Agent Details

The agent will be able to view his name and also his current status. 

![MyChatsScreen3](assets/AgentDetails.png)

You can also choose the change the status by clicking on **"Go Online"** or **"Go Offline"**. Before changing the status, a pop up would come to confirm the change.

![MyChatsScreen4](assets/offlinepopup.png)

> In the online state ---- WAITING FOR DEEP's TEXT

Reviewer: ```suggestion
Online State:
When an agent goes online,
-The agent will start receiving new chats as per his available slots.
-The agent could re-assign, mark as waiting, or complete any of the chats allocated to him

Offline State
When an agent goes Offline
- The agent will no longer receive any fresh chats
- The agent can no longer mark a chat as waiting
- All chats that were already marked as waiting by him/her will start assigning back to the agent so that they can close the chat, or re-assign the same
- The agent is expected to wrap up the chats that are already assigned to him

When an Agent Logsout
- All chats that are currently assigned to him/her will be auto completed.

```

Author: Added

ONLINE POP UP SCREENSHOT (Tanvi)

> An agent can receive new conversations only in the Online state.

#### Live Chat Stats

This section will provide all numbers related to incoming, existing and completed conversations taken by the agent logged in.

![MyChatsScreen5](assets/livechats.png)

- **Completed**: Number of chats completed today, as per the agent's timezone.

> - **Waiting for user**: Number of chats sent in waiting state by this agent.

Reviewer: ```suggestion
- **Waiting for user**: Number of chats currently in waiting state, as marked by this agent
```

> - **Agent queue**: Chats that have been manually reassigned to the agent and waiting chats where user has sent a response.

Reviewer: ```suggestion
- **Agent queue**: Chats that have been manually reassigned to the agent either by a fellow team mate or a team lead, or chats that were marked as waiting, and the user has replied back.  
Note: Chats get added to the agent queue, only if the agent is at max concurrency.
```

> - **Team Queue**: Chats that are currently waiting to be taken up by the team. 

Reviewer: ```suggestion
- **Team Queue**: Chats that are currently waiting to be taken up by the agents of team, basis their slot availability. 
```

> If an agent is part of multiple teams, we show the combined number for team queue.

#### Your Active Chats

This section will show all chats that are currently assigned to the agent.

![MyChatsScreen6](assets/chatslist.png)

We show the following details -

- **User name**: This is the name of the user. If the user has provided his name, that would show up or it will be shown as **Guest User**.

> - **Team name**: This is the name of the agent's team.

Reviewer: ```suggestion
- **Team name**: This is the name of the team to which this chat is allocated.
```

There are few tags which come on the chats depending on whether they have read or unread messages.

1. **Last messaged**: This will come up for read chats where agents have sent a response. 

2. **Message time**: This will either be the first user message time (among consecutive user messages) for unread chats, or the latest agent reply time for read chats, whichever is earlier.

> 3. **Waiting longest**: This is a tag used for unread chats which have not had a reply for over 5 mins, or the longest waiting chat.

Reviewer: ```suggestion
3. **Waiting longest**: This is a tag used for unread chats which have not had a reply for over 5 mins, or the longest waiting chat.

Note: For an agent, there will always be a maximum of one chat tagged as waiting longest.
```
